File: 7KYU/Simple string characters/solution.js
const uppercase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
const lowercase = "abcdefghijklmnopqrstuvwxyz";
const numbers = "0123456789";
function solve(str){
  var u = 0;
  var l = 0;
  var n = 0;
  var s = 0;
  for(var i = 0; i < str.length; i++) {
    if(uppercase.includes(str[i])) {
       u++;
    } else if(lowercase.includes(str[i])) {
      l++;
    } else if(numbers.includes(str[i])) {
      n++;
    } else {
      s ++;
    }   
  }
  return [u, l, n, s];
}

File: 7KYU/Simple string characters/solution.go
package kata

import "unicode"

func Solve(s string) []int {
	upperCase := 0
  lowerCase := 0
  numbers := 0
  special := 0
  
  for _, c := range s {
    if unicode.IsUpper(c) {
      upperCase +=1
    } else if unicode.IsLower(c) {
      lowerCase += 1
    } else if unicode.IsDigit(c) {
      numbers += 1
    } else {
      special += 1
    }
  }
  
  return []int{upperCase, lowerCase, numbers, special}
}